src/lsp/main.ts

```diff
@@ -377,65 +409,30 @@ async function startClient(): Promise<boolean> {
   updateStatusItem('starting');
 
   client.start();
-  await onReadyPromise.catch((e) => {
-    console.error('clojure-lsp:', e);
-    setStateValue(LSP_CLIENT_KEY_ERROR, e);
-  });
-  if (getStateValue(LSP_CLIENT_KEY_ERROR)) {
-    return false;
-  }
-  setStateValue(LSP_CLIENT_KEY, client);
-  const serverInfo = await getServerInfo(client);
-  if (serverInfo) {
-    serverVersion = serverInfo['server-version'];
-    sayClientVersionInfo(serverVersion, serverInfo);
+  try {
+    await onReadyPromise;
+    setStateValue(LSP_CLIENT_KEY, client);
+    const serverInfo = await getServerInfo(client);
+    if (serverInfo) {
+      serverVersion = serverInfo['server-version'];
+      sayClientVersionInfo(serverVersion, serverInfo);
+    }
+    client.onNotification('clojure/textDocument/testTree', (tree: TestTreeParams) => {
+      testTreeHandler(tree);
+    });
+    client.onRequest('window/showMessageRequest', handleShowMessageRequest);
     await vscode.commands.executeCommand('setContext', 'clojureLsp:active', true);
     updateStatusItem('started');
-  } else {
+    return true;
+  } catch (error) {
+    console.error('clojure-lsp:', error);
+    setStateValue(LSP_CLIENT_KEY_ERROR, error);
```

Comment:
Did you try this with a non-folder scenario, like with the Getting Started REPL? I ran into problems doing serverInfo there.